flow-decomposition/src/main/java/com/powsybl/flow_decomposition/glsk_provider/AutoGlskProvider.java

```diff
@@ -23,17 +23,23 @@
 public class AutoGlskProvider implements GlskProvider {
     @Override
     public Map<Country, Map<String, Double>> getGlsk(Network network) {
-        Map<Country, Map<String, Double>> glsks = network.getCountries().stream().collect(Collectors.toMap(
-            Function.identity(),
-            country -> new HashMap<>()));
-        network.getGeneratorStream()
-            .forEach(generator -> {
-                Country generatorCountry = NetworkUtil.getInjectionCountry(generator);
-                glsks.get(generatorCountry).put(generator.getId(), generator.getTargetP());
-            });
+        Map<Country, Map<String, Double>> glsks =
+                network.getCountries()
+                       .stream()
+                       .collect(Collectors.toMap(Function.identity(), country -> new HashMap<>()));
+
+        network.getGeneratorStream().filter(g -> g.getTerminal().isConnected()).forEach(generator -> {
+            Country generatorCountry = NetworkUtil.getInjectionCountry(generator);
+            glsks.get(generatorCountry).put(generator.getId(), generator.getTargetP());
+        });
+
         glsks.forEach((country, glsk) -> {
             double glskSum = glsk.values().stream().mapToDouble(factor -> factor).sum();
-            glsk.forEach((key, value) -> glsk.put(key, value / glskSum));
+            if (glskSum == 0.0) {
+                glsk.forEach((key, value) -> glsk.put(key, 1.0 / glsk.size()));
```

Comment:
Could you test this behavior with a basic network that has a null total target P please ?

Reply:
Thanks for the feedback, I added a couple of tests for that